istioctl/cmd/istioctl/istioctl_test.go

```diff
@@ -0,0 +1,416 @@
+			false,
+		},
+		{ // case 1
+			testRouteRules,
+			strings.Split("delete routerule a b", " "),
+			`Deleted config: routerule a
+Deleted config: routerule b
+`,
+			nil,
+			false,
+		},
+		{ // case 3 - delete by filename of istio config which doesn't exist
+			testRouteRules,
+			strings.Split("delete -f convert/testdata/v1alpha1/route-rule-80-20.yaml", " "),
+			"",
+			regexp.MustCompile("^Error: 1 error occurred:\n\n\\* cannot delete route-rule/default/route-rule-80-20: item not found\n$"),
+			true,
+		},
+	}
+
+	for i, c := range cases {
+		verifyOutput(t, fmt.Sprintf("TestGet case %d", i), c)
+	}
+}
+
+// mockClientFactoryGenerator creates a factory for model.ConfigStore preloaded with data
+func mockClientFactoryGenerator(configs []model.Config) func() (model.ConfigStore, error) {
+	outFactory := func() (model.ConfigStore, error) {
+		// Initialize the real client to get the supported config types
+		realClient, err := newClient()
+		if err != nil {
+			return nil, err
+		}
+
+		// Initialize memory based model.ConfigStore with configs
+		outConfig := memory.Make(realClient.ConfigDescriptor())
+		for _, config := range configs {
+			if _, err := outConfig.Create(config); err != nil {
+				return nil, err
+			}
+		}
+
+		// Wrap the memory ConfigStore so List() is sorted
+		return sortedConfigStore{store: outConfig}, nil
+	}
+
+	return outFactory
+}
+
+func (cs sortedConfigStore) Create(config model.Config) (string, error) {
+	return cs.store.Create(config)
+}
+
+func (cs sortedConfigStore) Get(typ, name, namespace string) (*model.Config, bool) {
+	return cs.store.Get(typ, name, namespace)
+}
+
+func (cs sortedConfigStore) Update(config model.Config) (string, error) {
+	return cs.store.Update(config)
+}
+func (cs sortedConfigStore) Delete(typ, name, namespace string) error {
+	return cs.store.Delete(typ, name, namespace)
+}
+
+func (cs sortedConfigStore) ConfigDescriptor() model.ConfigDescriptor {
+	return cs.store.ConfigDescriptor()
+}
+
+// List() is a facade that always returns cs.store items sorted by name/namespace
+func (cs sortedConfigStore) List(typ, namespace string) ([]model.Config, error) {
+	out, err := cs.store.List(typ, namespace)
+	if err != nil {
+		return out, err
+	}
+
+	// Sort by name, namespace
+	sort.Slice(out, func(i, j int) bool {
+		iName := out[i].ConfigMeta.Name
+		jName := out[j].ConfigMeta.Name
+		if iName == jName {
+			return out[i].ConfigMeta.Namespace < out[j].ConfigMeta.Namespace
+		}
+		return iName < jName
+	})
+
+	return out, nil
+}
+
+func verifyOutput(t *testing.T, label string, c testCase) {
+	// Override the client factory used by main.go
+	clientFactory = mockClientFactoryGenerator(c.configs)
+
+	var out bytes.Buffer
+	rootCmd.SetOutput(&out)
+	rootCmd.SetArgs(c.args)
+
+	file = "" // Clear, because we re-use
+
+	fErr := rootCmd.Execute()
+	output := out.String()
+
+	if c.expectedOutput != "" && c.expectedOutput != output {
+		t.Errorf("Unexpected output for %s %s: Expected\n%q,got\n%q",
```

Comment:
Use `t.Fatalf` throughout this function to abort the test case on error. You'll need to convert the caller (e.g. `TestDelete` to use [subtests](https://golang.org/pkg/testing/#T.Run).